src/calculator.js

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import numbersAndSymbols from "./arrayData"
+
+let arrayNumber = []
+let arrayNumberAndSigns = [] //este se puede usar en vez del otro, revisar 
+let newArrayStrings = []
+
+class CalculatorComponent extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      displayNumClicked: "",
+      displayNumAndOperations: ""
+    }
+    this.handleElementsClick = this.handleElementsClick.bind(this)
+    this.addNumberToArray = this.addNumberToArray.bind(this)
+    this.addNumberAndOperationsToArray = this.addNumberAndOperationsToArray.bind(this)
+    this.organizeInformation = this.organizeInformation.bind(this)
+    this.makeOperation = this.makeOperation.bind(this)
+  }
+
+  handleElementsClick(e){
+
+    let textDisplay = ""
+    let event = e.target.innerHTML 
+    this.addNumberAndOperationsToArray(event)
+    
+    if(event === 'AC'){
+      textDisplay = ''
+      arrayNumber = []
+    } else if(Number(event)){
+      let numClicked = Number(event)
+      let number = this.addNumberToArray(numClicked)
+      number = number.join('')
+      textDisplay = number    
+    } else if (event === '='){
+      textDisplay = this.makeOperation(newArrayStrings)
+      arrayNumber = []
+      arrayNumberAndSigns = []
+    } else {
+      //aca entra cuando agrego un simbolo
+      textDisplay = event
+      arrayNumber = []
+    }
+
+    this.setState({
+      displayNumClicked: textDisplay
+    })
+  }
+
+  addNumberToArray(numClicked){
+    //it adds the number I clicked to the array and displays it on the screen into display-number element 
+    arrayNumber.push(numClicked)
+    return arrayNumber
+  }
+
+  addNumberAndOperationsToArray(event){
+    //in this function i add all clicked numbers and signs to arrayNumberAndSigns array
+    if(event !== "AC"){
+      arrayNumberAndSigns.push(event)
+    } else {
+      arrayNumberAndSigns = []
+    }
+
+    this.setState({
+      displayNumAndOperations: arrayNumberAndSigns.join("")
+    })
+    this.organizeInformation(arrayNumberAndSigns)
+  }
+
+  organizeInformation(arrayNumberAndSigns){
+    newArrayStrings = [...arrayNumberAndSigns]
+
+    for(let counter = 0; counter < newArrayStrings.length; counter++){
+  
+      if((Number(newArrayStrings[counter]) &&  Number(newArrayStrings[counter + 1])) || (newArrayStrings[counter + 1] === '0')){
```

Comment:
* create a variable called `currentSymbol = newArrayStrings[counter]` 
* create a variable called `nextSymbol = newArrayStrings[counter + 1]`
use those variables instead of repeating that code every where